src/controllers/trip.controller.js

```diff
@@ -108,6 +108,34 @@ class TripController {
     ResponseService.setSuccess(200, 'List of all available locations', availableLocations);
     return ResponseService.send(res);
   }
+
+  /**
+ *
+ *
+ * @static
+ * @param {req} req
+ * @param {res} res
+ * @returns {response} @memberof Trips
+ */
+  static async editPendingTripRequest(req, res) {
+    const userToken = req.userData, userId = userToken.id, { tripId } = req.params;
+    const findTripRequest = await RequestService.findRequestByProperty({ requesterId: userId });
+    if ((findTripRequest) && (findTripRequest.tripId === tripId) && (findTripRequest.status === 'pending')) {
+      await Promise.all(Object.keys(req.body).map(async (key, index) => {
+        await TripService.updateTrip(
+          { userId, tripId },
+          { [key]: (Object.values(req.body)[index]) }
+        );
+      }));
+      await RequestService.updateRequest({ requesterId: userId, tripId }, { status: 'pending' });
```

Comment:
This method should return the updated row. There is no need to do another find on line 131